### ISSUE TITLE
Results in the paper

### ISSUE THREAD

**Issue by ZhuoranYu** [TorchSSL/TorchSSL#17]
Thanks for your great work! 

Just one quick question, how many GPUs were used to obtain the results in the paper? I didn't seem to find the specification on this.

Best

**Comment by ZhuoranYu** [comment]
And another follow-up question. I noticed that the results of baselines methods reported in your paper on Cifar-10 is substantially better than other papers. For example, Fixmatch with 40 labels in the original paper achieves 13.81 error rate whereas your results of fixmatch only get 6.78 error rate with 40 labels. 

Why is that? Did I miss anything important?

Thanks for your time!

**Comment by 18445864529** [comment]
Sorry, we didn't have a deep look at Google's TensorFlow repo (which gives the 13.81 result) so I cannot pinpoint the reason, it could be the framework or some implementation difference. 
Besides, currently please do not completely refer to the result in the paper as there exist some inaccurate results (e.g. for cifar100-400) and we'll update the results in the paper soon. Please take the result by running this repo as the reference.

**Comment by houwenxin** [comment]
Hi, thank you for your interest in our work! 

We used 1 GPU for training on cifar10 & svhn, 2 GPUs on stl and cifar100, 4 GPUs on imagenet. Hope this answers your question.

**Comment by ZhuoranYu** [comment]
Thanks for your reply! Actually, I think the difference between your results and original FixMatch paper is non-trivial. It would be great if you can check for this if you have time. Your results are also much higher than other pytorch implementations such as https://github.com/LeeDoYup/FixMatch-pytorch.

 I'm also taking a look at your implementation. I'll leave the issue open for now and come back if I find any specific difference so that we can better understand the problem. 

Again, thanks for your work!

**Comment by qianlanwyd** [comment]
Hi, we updated the results. Please check.

**Comment by machengcheng2016** [comment]
> Thanks for your reply! Actually, I think the difference between your results and original FixMatch paper is non-trivial. It would be great if you can check for this if you have time. Your results are also much higher than other pytorch implementations such as https://github.com/LeeDoYup/FixMatch-pytorch.
> 
> I'm also taking a look at your implementation. I'll leave the issue open for now and come back if I find any specific difference so that we can better understand the problem.
> 
> Again, thanks for your work!

Greetings, have you figured out the reason of performance gap between `TorchSSL` and `FixMatch-pytorch`? It really confuses a lot! Thank you!

**Comment by machengcheng2016** [comment]
> > Thanks for your reply! Actually, I think the difference between your results and original FixMatch paper is non-trivial. It would be great if you can check for this if you have time. Your results are also much higher than other pytorch implementations such as https://github.com/LeeDoYup/FixMatch-pytorch.
> > I'm also taking a look at your implementation. I'll leave the issue open for now and come back if I find any specific difference so that we can better understand the problem.
> > Again, thanks for your work!
> 
> Greetings, have you figured out the reason of performance gap between `TorchSSL` and `FixMatch-pytorch`? It really confuses a lot! Thank you!

Never mind. I found the reason :)
